tools/myst/enc/enc.c

```diff
@@ -153,18 +156,102 @@ static size_t _get_num_tcs(void)
 
 int myst_setup_clock(struct clock_ctrl*);
 
-/* Handle illegal SGX instructions */
-static uint64_t _vectored_handler(oe_exception_record_t* er)
+static void _sanitize_xsave_area_fields(uint64_t* rbx, uint64_t* rcx)
 {
+    assert(rbx && rcx);
+    /* replace XSAVE/XRSTOR save area size with fixed large value of 4096,
+    to protect against spoofing attacks from untrusted host.
+    If host returns smaller xsave area than required, this can cause a buffer
+    overflow at context switch time.
+    We believe value of 4096 should be sufficient for forseeable future. */
+    if (*rbx < 4096)
+        *rbx = 4096;
+    if (*rcx < 4096)
+        *rcx = 4096;
+}
+
+#define COLOR_GREEN "\e[32m"
+#define COLOR_RESET "\e[0m"
+
 #define RDTSC_OPCODE 0x310F
 #define CPUID_OPCODE 0xA20F
 #define IRETQ_OPCODE 0xCF48
 #define SYSCALL_OPCODE 0x050F
 
+typedef struct _opcode_pair
+{
+    long opcode;
+    const char* str;
+} opcode_pair_t;
+
+static opcode_pair_t _opcode_pairs[] = {
+    {RDTSC_OPCODE, "rdtsc"},
+    {CPUID_OPCODE, "cpuid"},
+    {IRETQ_OPCODE, "iretq"},
+    {SYSCALL_OPCODE, "syscall"},
+};
+
+static size_t _n_pairs = sizeof(_opcode_pairs) / sizeof(_opcode_pairs[0]);
+const char* opcode_str(long n)
+{
+    for (size_t i = 0; i < _n_pairs; i++)
+    {
+        if (n == _opcode_pairs[i].opcode)
+            return _opcode_pairs[i].str;
+    }
+
+    return "unknown";
+}
+
+__attribute__((format(printf, 2, 3))) static void _exception_handler_strace(
+    long n,
+    const char* fmt,
+    ...)
+{
+    if (_trace_syscalls)
```

Comment:
The release_build PR tries to consolidate strace switches. @mikbras , please review this code.

Reply:
Vectored exception handler tracing is now consistent with syscall tracing, its turned off if `tee_debug_mode` is not set.